odl/solvers/scalar/steplen.py

```diff
@@ -148,6 +148,21 @@ def __call__(self, x, direction, dir_derivative):
             The computed step length
         """
         alpha = 1.0
+
+        # Since all comparisons with nan returns False, in order to not get a
+        # problem with nan-values we throw an error.
+        if self.function(x) == float('nan'):
```

Comment:
This is critical to do, as of now it would require two calls, often doubling execution time.